app/src/main/java/io/github/droidkaigi/confsched2018/presentation/sponsors/SponsorsFragment.kt

```diff
@@ -79,8 +87,16 @@ class SponsorsFragment : Fragment(), Injectable {
             spanCount = SPONSOR_SCREEN_MAX_SPAN_SIZE
             add(sponsorPlansSection)
             setOnItemClickListener({ item, _ ->
-                val url = (item as? SponsorItem)?.sponsor?.link ?: return@setOnItemClickListener
+                val sponsor = (item as? SponsorItem)?.sponsor
+                val url = sponsor?.link ?: return@setOnItemClickListener
                 navigationController.navigateToExternalBrowser(url)
+                val params = Bundle()
+                params.putString(FirebaseAnalytics.Param.ITEM_NAME, sponsor.link)
+                params.putString(FirebaseAnalytics.Param.CONTENT_TYPE,
+                        getString(R.string.firebase_analytics_event_sponsor))
+                params.putString(FirebaseAnalytics.Param.GROUP_ID,
+                        item.planType::class.java.simpleName.toLowerCase())
+                fireBaseAnalytics.logEvent(FirebaseAnalytics.Event.SELECT_CONTENT, params)
```

Comment:
Just suggestion 🙋 
How about creating `EventTracker` or `FirebaseEvent` class and inject to ViewModel constructor by Dagger?

Reply:
@konifar 
Sorry for my late reply.
I fixed it, but I have a problem.
I  was going to move code which send page view to FirebaseEvent too, and call it from viewModel, but Fragment#setUserVisibleHint is called before onCreate, so viewModel doesn’t initialize and  UninitializedPropertyAccessException happens.
Do you have any good idea?I just pushed code above. 90911f3

Reply:
Thanks for trying to fix it. Understood. But I don't have any good idea for now 🙇 
Can you revert https://github.com/DroidKaigi/conference-app-2018/commit/90911f3d37579a7dcd213a845a4572068e3599b6 ? After that, I'll merge your PR!

Reply:
OK!